python_modules/dagster/dagster/_core/definitions/asset_check_factories/freshness_checks/sensor.py

```diff
@@ -155,7 +122,70 @@ def ordered_iterator_freshness_checks_starting_with_key(
     # Offset based on the left off asset check key, but then iterate back through the beginning afterwards
     if left_off_asset_check_key:
         left_off_idx = asset_check_keys_sorted.index(left_off_asset_check_key)
-        yield from asset_check_keys_sorted[left_off_idx:]
-        yield from asset_check_keys_sorted[:left_off_idx]
+        yield from asset_check_keys_sorted[left_off_idx + 1 :]
```

Comment:
why +1 now?

Reply:
realized that this method of indexing was actually inclusive; meaning we would start iterating at the item we evaluated the previous run of the sensor, rather than the item __after__ the last one we evaluated. this fixes (and I added a test for the ordering behavior)